### DISCUSSION TITLE
card draft

### DISCUSSION THREAD

**Discussion by MohamedElashri** [JordanKnott/taskcafe#60] [Ideas]
**Is your feature request related to a problem? Please describe.**
A good feature in Trello and even some small task manager self-hosted solutions like Planka is that ability to draft cards if not added so that it does not get discarded if you click anywhere 

**Describe the solution you'd like**
The solution will be just adding a card draft feature when adding a new card

**Comment by JordanKnott** [comment]
I had thought about implementing this when I was working on the card composer but ended up liking that it just cancels the task name when clicking away.

I'll give this some thought - I might just switch it to work the way you described or maybe add it as an option?

**Comment by JordanKnott** [comment]
After some thought - Im going to go ahead and implement this. The current plan is to include it in the next 0.3.0 release.

**Comment by MohamedElashri** [comment]
I think this is better as I think it is a very reliable feature. personally I moved to use taskcafe and sometimes I click by mistake on other parts and lose whatever I wrote.

**Comment by JordanKnott** [comment]
I have marked this task as a good first issue.

I would personally implement the "draft" state in the Board component (`frontend/src/Projects/Project/Board/index.tsx`).

Relevant files

- `frontend/src/Projects/Project/Board/index.tsx`
- `frontend/src/shared/components/Card/index.tsx`
- `frontend/src/shared/components/Lists/index.tsx`

Please take a look at the [developers guide](https://github.com/JordanKnott/taskcafe/wiki/Developers-Guide) for an idea on how the project is structured.
